src/components.js

```diff
@@ -21,6 +22,16 @@ export const Nav = () => `
   </nav>
   `;
 
+const ids = (args) => {
+  let ans = '';
+  for (const [key, value] of Object.entries(args)) {
+    if (value != null) {
+      ans = `${key}=${value}`;
+    }
+  }
+  return ans;
+};
+
```

Comment:
I am not sure what this function is doing, but the component.js is for just components. Let's try to keep it clean

Reply:
The initial id you passed to the buttons was not the actual id needed to query the API the one needed is an object of three entries and some of the entries are usually null that's why I have to check for the valid entry and use it for the fetch